src/librustc_metadata/creader.rs

```diff
@@ -763,6 +764,53 @@ impl<'a> CrateLoader<'a> {
                                   &|data| data.needs_panic_runtime());
     }
 
+    fn inject_sanitizer_runtime(&mut self) {
+        // HACK we don't want to sanitize build scripts ...
+        if self.local_crate_name == "build_script_build" {
+            return;
+        }
+
+        if let Some(ref sanitizer) = self.sess.opts.debugging_opts.sanitizer {
+            // Sanitizers can only be used with executables linked to `std`
+            if !self.sess.crate_types.borrow().iter().all(|ct| {
+                match *ct {
+                    config::CrateTypeExecutable => true,
+                    _ => false
+                }
+            }) {
+                return
```

Comment:
As we discussed during triage, could this actually generate an error? Both for non-executables but also for unsupported platforms.